cpp/src/IceStorm/TopicManagerI.cpp

```diff
@@ -63,76 +63,76 @@ class TopicManagerI : public TopicManagerInternal
             else
             {
                 FinishUpdateHelper unlock(_instance->node());
-                return _impl->create(id);
+                return _impl->create(move(id));
             }
         }
     }
 
-    virtual TopicPrx retrieve(const string& id, const Ice::Current&) const
+    shared_ptr<TopicPrx> retrieve(string id, const Ice::Current&) override
     {
         // Use cached reads.
         CachedReadHelper unlock(_instance->node(), __FILE__, __LINE__);
-        return _impl->retrieve(id);
+        return _impl->retrieve(move(id));
     }
 
-    virtual TopicDict retrieveAll(const Ice::Current&) const
+    TopicDict retrieveAll(const Ice::Current&) override
     {
         // Use cached reads.
         CachedReadHelper unlock(_instance->node(), __FILE__, __LINE__);
         return _impl->retrieveAll();
     }
 
-    virtual NodePrx getReplicaNode(const Ice::Current&) const
+    shared_ptr<NodePrx> getReplicaNode(const Ice::Current&) const override
     {
         // This doesn't require the replication to be running.
         return _instance->nodeProxy();
     }
 
 private:
 
-    TopicManagerPrx getMaster(Ice::Long& generation, const char* file, int line) const
+    shared_ptr<TopicManagerPrx> getMaster(Ice::Long& generation, const char* file, int line) const
```

Comment:
(Minor) Ice::Long is the C++98 mapping for Slice long. The C++11/17 mapping for Slice long is `long long`.